src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -264,6 +264,14 @@ public async Task WaitForDependenciesAsync(IResource resource, CancellationToken
         var pendingDependencies = new List<Task>();
         foreach (var waitAnnotation in waitAnnotations)
         {
+            if (waitAnnotation.Resource is ParameterResource or ResourceWithConnectionStringSurrogate)
+            {
+                // Parameters and connection string resources are inert and don't need to be waited on.
+                // If we add support for parameter resources that can be waited on, we can remove this check.
+                // As of right now, we don't support waiting on parameter resources.
+                continue;
```

Comment:
Write a warning log message? Let people know they're doing it wrong.

Reply:
Can't yet. Take a look at the issue, this pattern is from our docs.